core/src/main/java/org/frankframework/http/mime/MultipartUtils.java

```diff
@@ -110,4 +200,95 @@ private static String parseParameterField(String cdFields, String fieldName) {
 		}
 		return null;
 	}
+
+	public static record MultipartMessages(Message multipartXml, Map<String, Message> messages) {}
+
+	public static MultipartMessages parseMultipart(InputStream inputStream, String contentType) throws IOException {
+		try (InputStream unused = inputStream) {
+			final InputStreamDataSource dataSource = new InputStreamDataSource(contentType, inputStream); //the entire InputStream will be read here!
+			final MimeMultipart mimeMultipart = new MimeMultipart(dataSource);
+			final XmlBuilder attachments = new XmlBuilder("parts");
+			final Map<String, Message> parts = new LinkedHashMap<>();
+
+			for (int i = 0; i < mimeMultipart.getCount(); i++) {
+				final BodyPart bodyPart = mimeMultipart.getBodyPart(i);
+				final String fieldName = getFieldName(bodyPart);
+				if(StringUtils.isEmpty(fieldName)) {
+					log.info("unable to determine fieldname skipping part");
+					continue;
+				}
+
+				final XmlBuilder attachment = new XmlBuilder("part");
+				attachment.addAttribute("name", fieldName);
+				PartMessage message = new PartMessage(bodyPart);
+				parts.put(fieldName, message);
+				if (!isBinary(bodyPart)) {
+					// Process regular form field (input type="text|radio|checkbox|etc", select, etc).
+					log.trace("setting multipart formField [{}] to [{}]", fieldName, message);
+					attachment.addAttribute("type", "text");
+					attachment.addAttribute("value", message.asString());
+				} else {
+					// Process form file field (input type="file").
+					final String fileName = getFileName(bodyPart);
+					log.trace("setting parameter [{}] to input stream of file [{}]", fieldName, fileName);
+
+					attachment.addAttribute("type", "file");
+					attachment.addAttribute("filename", fileName);
+					attachment.addAttribute("size", message.size());
+					attachment.addAttribute("sessionKey", fieldName);
+					attachment.addAttribute("mimeType", extractMimeType(bodyPart.getContentType()));
+				}
```

Comment:
Wellicht kan je dit blok in een aparte method `getAttachment(BodyPart bodyPart)` oid kwijt? Dat leest wat makkelijker en mogelijk kan je dan de for loop ook vervangen door een java stream met .map?